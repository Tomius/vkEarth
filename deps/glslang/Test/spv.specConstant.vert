#version 400

layout(constant_id = 17) const int arraySize = 12;
in vec4 ucol[arraySize];

layout(constant_id = 17) const bool spBool = true;
layout(constant_id = 18) const float spFloat = 3.14;
layout(constant_id = 19) const double spDouble = 3.1415926535897932384626433832795;
layout(constant_id = 22) const uint scale = 2;

out vec4 color;
out int size;

void main()
{
    color = ucol[2];
    size = arraySize;
    if (spBool)
        color *= scale;
    color += float(spDouble / spFloat);
}
