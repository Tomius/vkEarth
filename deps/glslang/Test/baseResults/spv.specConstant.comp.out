spv.specConstant.comp
Warning, version 450 is not yet complete; most version-specific features are present, but some are missing.


Linked compute stage:


// Module Version 10000
// Generated by (magic number): 80001
// Id's are bound by 19

                              Capability Shader
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              EntryPoint GLCompute 4  "main"
                              ExecutionMode 4 LocalSize 32 32 1
                              Source GLSL 450
                              Name 4  "main"
                              Name 7  "bn"
                              MemberName 7(bn) 0  "a"
                              Name 9  "bi"
                              MemberDecorate 7(bn) 0 Offset 0
                              Decorate 7(bn) BufferBlock
                              Decorate 9(bi) DescriptorSet 0
                              Decorate 18 BuiltIn WorkgroupSize
               2:             TypeVoid
               3:             TypeFunction 2
               6:             TypeInt 32 0
           7(bn):             TypeStruct 6(int)
               8:             TypePointer Uniform 7(bn)
           9(bi):      8(ptr) Variable Uniform
              10:             TypeInt 32 1
              11:     10(int) Constant 0
              12:      6(int) Constant 1024
              13:             TypePointer Uniform 6(int)
              15:             TypeVector 6(int) 3
              16:      6(int) Constant 32
              17:      6(int) Constant 1
              18:   15(ivec3) ConstantComposite 16 16 17
         4(main):           2 Function None 3
               5:             Label
              14:     13(ptr) AccessChain 9(bi) 11
                              Store 14 12
                              Return
                              FunctionEnd
